src/commands/selectCommandTemplate.ts

```diff
@@ -0,0 +1,157 @@
+    return await selectCommandTemplate(
+        context,
+        'logs',
+        matchContext,
+        undefined,
+        { 'containerId': containerId }
+    );
+}
+
+export async function selectComposeCommand(context: IActionContext, matchContext: string[], folder: vscode.WorkspaceFolder, composeCommand: 'up' | 'down', configurationFile?: string, detached?: boolean, build?: boolean): Promise<string> {
+    return await selectCommandTemplate(
+        context,
+        composeCommand === 'up' ? 'composeUp' : 'composeDown',
+        matchContext,
+        folder,
+        { 'configurationFile': configurationFile ? `-f \"${configurationFile}\"` : '', 'detached': detached ? '-d' : '', 'build': build ? '--build' : '' }
+    );
+}
+
+async function selectCommandTemplate(context: IActionContext, command: TemplateCommand, matchContext?: string[], folder?: vscode.WorkspaceFolder, additionalVariables?: { [key: string]: string }): Promise<string> {
+    // Get the templates from settings
+    const config = vscode.workspace.getConfiguration('docker');
+    const templateSetting: CommandTemplate[] | string = config.get(`commands.${command}`);
+    let templates: CommandTemplate[];
+
+    // Get template(s) from settings
+    if (typeof (templateSetting) === 'string') {
+        templates = [{ template: templateSetting }] as CommandTemplate[];
+    } else {
+        templates = templateSetting;
+    }
+
+    // Look for settings-defined template(s) with explicit match, that matches the context
+    const matchedTemplates = templates.filter(template => {
+        if (template.match) {
+            try {
+                const matcher = new RegExp(template.match, 'i');
+                return matchContext.some(m => matcher.test(m));
+            } catch {
+                // Don't wait
+                // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                ext.ui.showWarningMessage(`Invalid match expression for template '${template.label}'. This template will be skipped.`);
+            }
+        }
+
+        return false;
+    });
+
+    // Look for settings-defined template(s) with no explicit match
+    const universalTemplates = templates.filter(template => !template.match);
+
+    // Select from explicit match templates, if none then from settings-defined universal templates, if none then hardcoded default
+    let selectedTemplate: CommandTemplate;
+    if (matchedTemplates.length > 0) {
+        selectedTemplate = await quickPickTemplate(context, matchedTemplates);
+    } else if (universalTemplates.length > 0) {
+        selectedTemplate = await quickPickTemplate(context, universalTemplates);
+    } else {
+        selectedTemplate = defaults[command];
+    }
+
+    context.telemetry.properties.isDefaultCommand = selectedTemplate.template === defaults[command].template ? 'true' : 'false';

```

Comment:
Note this would be `true` both if the user did not customize the command (and is using the default command) and if the user *did* customize the command but happened to set it to the same value as the default command.  I'm not sure if we care about that distinction.

Reply:
Yeah, that was intentional on my part, as it is not possible to distinguish whether the setting value came from extension defaults, or if it was a user-provided value that happened to be identical to extension defaults. I don't imagine many users will be providing values that are identical to defaults as that is redundant.